cmd/nodepool/core/create.go

```diff
@@ -142,3 +142,24 @@ func (o *CreateNodePoolOptions) CreateNodePool(ctx context.Context, platformOpts
 	fmt.Printf("NodePool %s created\n", o.Name)
 	return nil
 }
+
+// ValidateHostedClusterPayloadSupportsNodePoolCPUType validates the HostedCluster payload type can support the CPU architecture
+// of the NodePool.
+func ValidateHostedClusterPayloadSupportsNodePoolCPUType(ctx context.Context, client crclient.Client, name, namespace, arch string) error {
+	logger := ctrl.LoggerFrom(ctx)
+
+	hc := &hyperv1.HostedCluster{}
+	err := client.Get(context.Background(), types.NamespacedName{Namespace: namespace, Name: name}, hc)
+	if err != nil {
+		// This is expected to happen when we create a cluster since there is no created HostedCluster CR to check the
+		// payload from.
+		logger.Info("WARNING: failed to get HostedCluster to check payload type")
+		return nil
+	}
+
+	if hc.Status.PayloadArch != "" && hc.Status.PayloadArch != hyperv1.Multi && hc.Status.PayloadArch != hyperv1.ToPayloadArch(arch) {
```

Comment:
This has a race for when hc.Status.PayloadArch is not set yet. Can we account for that use case and let it succeed with a warning?

Reply:
Updated